src/screens/GroupSetting/hooks/useGroupSetting.js

```diff
@@ -54,16 +55,16 @@ const useGroupSetting = ({route}) => {
       setUrlImage(selectedChannel?.channelPicture);
     }
   }, [selectedChannel]);
-
   const submitData = async (withNavigation = true) => {
-    if (!changeImage || changeName) return navigation.goBack();
+    if (!changeImage && !changeName) return navigation.goBack();
     try {
       setLoadingUpdate(true);
       let body = {
         channel_id: route?.params?.id,
-        channel_name: groupName
+        channel_name: groupName,
+        channel_image: urlImage
       };
-      if (urlImage) {
+      if (urlImage && !isValidUrl(urlImage)) {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic in this code block seems incorrect. The condition `if (urlImage && !isValidUrl(urlImage))` checks if `urlImage` is not empty and is not a valid URL, then tries to upload the image. However, if the URL is invalid, the upload should not proceed. Also, the `channel_image` property is set to `urlImage` without checking its validity first. Consider revising this logic.

```diff
-  let body = {
-    channel_id: route?.params?.id,
-    channel_name: groupName,
-    channel_image: urlImage
-  };
-  if (urlImage && !isValidUrl(urlImage)) {
-    try {
-      const res = await uploadPhotoImage(urlImage);
-      if (res) {

+  let body = {
+    channel_id: route?.params?.id,
+    channel_name: groupName,
+  };
+  if (urlImage && isValidUrl(urlImage)) {
+    body.channel_image = urlImage;
+    try {
+      const res = await uploadPhotoImage(urlImage);
+      if (res) {
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->